prow/job/artifacts.py

```diff
@@ -0,0 +1,177 @@
+                test_count += test_summary.tests
+                if test_summary.failures > 0:
+                    failed_tests.extend(report.get_failed_tests())
+        else:
+            raise FileNotFoundError("Cannnot find any junit file")
+
+        test_failures_summary = {
+            "ID": int(self._job_run_id),
+            "ProwJob": {
+                "Name": self._job_name
+            },
+            "Tests": failed_tests,
+            "TestCount": test_count
+        }
+
+        return json.dumps(test_failures_summary).encode('utf-8')
+
+
+class GCSClient():
+
+    def __init__(self, cred_file, bucket):
+        # check cred file exist or not
+        if not os.path.exists(cred_file):
+            raise FileNotFoundError(f"file {cred_file} does not exist")
+        # init storage client
+        cred = Credentials.from_service_account_file(cred_file)
+        self._client = storage.Client(credentials=cred)
+        # get root directory
+        self._bucket = self._client.bucket(bucket)
+
+    def get_file(self, path):
+        # if we know the absolute file path, get the blob directly
+        return self._bucket.get_blob(path)
+
+    def get_files(self, path, name_patterns: list = None):
+
+        blobs = self._bucket.list_blobs(prefix=path)
+        if name_patterns:
+            matched_blobs = []
+            for b in blobs:
+                for p in name_patterns:
+                    if re.compile(p).search(b.name):
+                        matched_blobs.append(b)
+                        break
+            return matched_blobs
+        else:
+            return blobs
+
+
+class JunitTestReport():
+
+    def __init__(self, file_content: bytes = None, file_path: str = None):
+        if file_content is None and file_path is None:
+            raise ValueError("all the arguments are empty")
+
+        if file_content:
+            self._root = ElementTree.fromstring(file_content.decode("utf-8"))
+        else:
+            if os.path.exists(file_path):
+                self._root = ElementTree.parse(file_path).getroot()
+            else:
+                raise FileNotFoundError(f"{file_path} not found")
+
+    def get_test_summary(self):
+        return JunitTestSummary(self._root)
+
+    def get_failed_tests(self):
+        test_cases = []
+        elements = self._root.findall(".//testcase")
+        if elements:
+            for e in elements:
+                tc = JunitTestCase(e)
+                if tc.is_failure():
+                    # https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/logs/periodic-ci-openshift-release-master-nightly-4.15-e2e-aws-ovn-upi/1776477546024538112/artifacts/e2e-aws-ovn-upi/openshift-e2e-test/artifacts/junit/test-failures-summary_20240406-055752.json
+                    test_cases.append({
+                        "Test": {"Name": tc.name},
+                        "Suite": {"Name": "openshift-tests-private"},
```

Comment:
@rioliu-rh note: it is not correct. maybe different suite per differernt case type.
anyway, keep it now to change it later.